Makefile

```
@@ -7,9 +7,15 @@ CONTROLLER = $(OUTPUT_DIR)/bin/shipwright-build-controller
 
 # golang cache directory path
 GOCACHE ?= $(shell echo ${PWD})/$(OUTPUT_DIR)/gocache
+
 # golang target architecture
-GO_OS ?= $(shell uname | tr '[:upper:]' '[:lower:]')
+# Check if GO_OS is defined, if not set it to `linux` which is the only supported OS
+# this provides flexibility to users to set GO_OS in the future if require
+ifeq ($(origin GO_OS), undefined)
+GO_OS = "linux"
+endif
```

I do not have a Makefile black belt, but is not this the same:

```suggestion
GO_OS ?= "linux"
```

Reference: http://www.gnu.org/software/make/manual/html_node/Setting.html

---

its the same, see my commit msg on the usage of origin.

---

I see.

> I added the usage of the origin function on purpose, to make the logic very clear.

Hm, okay, my subjective view is that I can better understand `?=`, also as it exists in [other languages as well](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment).

But, let's be pragmatic, it works.